cmd/cloud/deployment.go

```diff
@@ -181,6 +183,7 @@ func newDeploymentUpdateCmd(out io.Writer) *cobra.Command {
 	cmd.Flags().StringVarP(&deploymentName, "deployment-name", "", "", "Name of the deployment to update")
 	cmd.Flags().StringVarP(&dagDeploy, "dag-deploy", "", "", "Enables DAG-only deploys for the deployment")
 	cmd.Flags().BoolVarP(&cleanOutput, "clean-output", "c", false, "clean output to only include inspect yaml or json file in any situation.")
+	cmd.Flags().StringVarP(&deploymentEnforceCD, "enforce-cicd", "", "", "Provide this flag either ON/OFF. ON means deploys to deployment must use an API Key or Token. This essentially forces Deploys to happen through CI/CD")
```

Comment:
why not a boolean flag? It would keep things consistent with other such flags we have in CLI.
If it is just about figuring out user-supplied value vs default value, then we can use `cmd.Flags().Changed("enforce-cicd")` for that

Reply:
issues with boolean flag is during update, even though i am not touching this flag, it would pick the default false value.

`cmd.Flags().Changed("enforce-cicd")` will also give a false value for Update function in cloud/deployment/deployment.go